mountpoint-s3/src/prefetch.rs

```diff
@@ -105,8 +108,10 @@ where
 }
 
 pub type ParquetPrefetcher<Runtime> = Prefetcher<ParquetPartStream<Runtime>>;
-pub type MetadataRef = Arc<AsyncRwLock<Option<ParsedMetadata>>>;
+pub type RowgroupColRanges = HashMap<(RowGroupIndex, ColumnIndex), Range<u64>>;
+pub type MetadataRef = Arc<AsyncRwLock<Option<(ParsedMetadata, RowgroupColRanges)>>>;
 pub type RawMetadataRef = Arc<AsyncRwLock<Option<RawMetadata>>>;
+type CachedRef = Arc<DashMap<ObjectId, (RawMetadataRef, MetadataRef, InMemoryCacheRef)>>;
```

Comment:
All these type aliases should really be refactored into structs. But besides that, it is not clear to me what is actually mutable (and when). For example, can the metadata for an `ObjectId` change?